packages/components/src/components/form/Select/index.tsx

```diff
@@ -310,17 +307,20 @@ export const Select = ({
                 searchedTerm.current += charValue;
             }
 
-            const { options } = selectRef.current.select.props;
+            const options = selectRef.current.props.options as OptionsOrGroups<
```

Comment:
better to fix the `isOptionGrouped` to make [actual type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) from it

Reply:
30fff942a1315acbb2d457f3bf5fd381ac0aa811